### DISCUSSION TITLE
--no-content-hash flag stlil hashes files

### DISCUSSION THREAD

**Discussion by HappyTiptoe** [parcel-bundler/parcel#6032] [Q&A] [answered]
I'm pretty new to discussions, and issues, on Github, so please direct me to the right place if I'm posting in the wrong place.

Basically, as the title says, when I run:

```
"build:simple": "parcel build ./src/index.html --no-optimize --no-content-hash --no-source-maps"
```

Inside `dist/`, the files are stlll hashed (e.g. `index.fi59gj4.js`). Is this a bug, or am I simply doing something wrong?

**Comment by mischnic** [comment, marked as answer]
They are still hashed but the hash doesn't change anymore when the bundle contents change.

I've tried to make that clearer in the docs: https://github.com/parcel-bundler/website/pull/839

**Reply by HappyTiptoe** [reply]
Understood, thanks mischnic!

Is there a way to perform my expected behavior of having no hashes, in v2 currently?

**Reply by mischnic** [reply]
No. But a third party namer plugin could certainly do this

**Reply by rrrnld** [reply]
I just stumbled over this. I was confused by the naming as well, but it's good to know that in effect it's simarly useful as not hashing at all. Maybe it makes sense to rename the flag to `--stable-bundle-names` or `--constant-file-names` or something like that?

**Reply by mischnic** [reply]
The names aren't necessarily stable (e.g. adding imports to files might change the hash)

It's just that the file content itself doesn't contribute to the hash, therefore `--no-content-hash`

**Comment by snnsnn** [comment]
I got `unknown option` error when I use it with `"parcel": "^2.0.1"`:

```bash
parcel src/index.html --dist-dir output --no-content-hash
error: unknown option '--no-content-hash'
```
